tests/gui_tests/test_project_downloader.py

```
@@ -0,0 +1,2304 @@
+
+    # Mock progress update to raise exception
+    with patch.object(downloader, "_update_progress") as mock_update:
+        mock_update.side_effect = Exception("Progress update failed")
+
+        # Should not raise exception
+        downloader.cancel_download()
+
+        assert downloader._cancelled is True
+
+
+def test_update_progress_exception_handling():
+    """Test _update_progress exception handling."""
+    downloader = ProjectDownloader(project="test-project")
+    downloader._total_files = 10
+    downloader._completed_files = 3
+    downloader._progress_callback = MagicMock(side_effect=Exception("Callback failed"))
+
+    # Should not raise exception
+    downloader._update_progress("downloading", "Test message")
+
+
+def test_report_error_exception_handling():
+    """Test _report_error exception handling."""
+    downloader = ProjectDownloader(project="test-project")
+    downloader._total_files = 10
+    downloader._completed_files = 3
+    downloader._progress_callback = MagicMock(side_effect=Exception("Callback failed"))
+    downloader._error_callback = MagicMock(side_effect=Exception("Error callback failed"))
+
+    # Should not raise exception
+    downloader._report_error("Test error")
+
+
+def test_context_manager_exit_exception():
+    """Test context manager exit with exception."""
+    with patch(
+        "dds_cli.dds_gui.pages.project_actions.download_data.project_downloader.dds_cli.directory.DDSDirectory"
+    ):
+        with patch(
+            "dds_cli.dds_gui.pages.project_actions.download_data.project_downloader.dds_cli.data_getter.DataGetter"
+        ):
+            with ProjectDownloader(project="test-project") as downloader:
+                # Mock cleanup to raise exception
+                with patch.object(downloader, "cleanup", side_effect=Exception("Cleanup failed")):
+                    # Should not raise exception
+                    pass  # Context manager should handle the exception
```

not sure why this test just passes, isn't it possible to check  that the exception is suppressed or logged?

---

Fixed test to verify exception suppression and logging: updated __exit__ to catch/log cleanup exceptions, and test now asserts both that exceptions are suppressed (no exception raised) and that they're logged via LOG.warning.